### PR TITLE
Argument parsing

### PR DESCRIPTION
**Pull request by wilmhit**

### CODE REVIEW

File: graph-coloring/__main__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import argparse
+from files import *
```
Comment (songofroland) [inline graph-coloring/__main__.py:2]: In general it's a bad practice. It's better to import everything explicitly

Also, the name _files_ sound for me like a 3rd party package, let's rename it.

File: graph-coloring/__main__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+def main(args):
+    graph_file = get_graph_file(args.GRAPH)
+    output = process_graph(graph_file)
+    save_png(output, args.out)
+
+def process_graph(graph) -> None: pass
+
+parser = argparse.ArgumentParser()
+parser.add_argument('GRAPH', type=str, help="Text file containing graph")
```
Comment (songofroland) [inline graph-coloring/__main__.py:12]: so this will go like `script.py GRAPH my_file.txt` ?

File: graph-coloring/__main__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import argparse
+from files import *
+
+def main(args):
+    graph_file = get_graph_file(args.GRAPH)
+    output = process_graph(graph_file)
+    save_png(output, args.out)
+
+def process_graph(graph) -> None: pass
```
Comment (songofroland) [inline graph-coloring/__main__.py:9]: Don't take it personal but it hurts my eyes. When reading this I feel that the function is incorrect. Let's go with:
```
def process_graph(graph) -> None: 
    pass
```

File: graph-coloring/__main__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import argparse
+from files import *
+
+def main(args):
```
Comment (songofroland) [inline graph-coloring/__main__.py:4]: please add type to `args`

File: graph-coloring/files.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+def get_graph_file(filename: str) -> str: pass
```
Comment (songofroland) [inline graph-coloring/files.py:2]: Same here

File: reqirements.dev [hunk 6]
```diff
@@ -0,0 +1 @@
+yapf==0.31.0
```
Comment (songofroland) [inline reqirements.dev:1]: Typo in filename. Also let's better use extension `.txt`.  (for IDE's it's better)

Ideally, let's use `poetry` or `pipenv` but it does not matter that much.

**Review by songofroland — CHANGES_REQUESTED** [review]

File: graph-coloring/__main__.py (hunk 2 above)
Comment (wilmhit) [reply graph-coloring/__main__.py:12]: no script.py myfile.txt

File: reqirements.dev (hunk 6 above)
Comment (wilmhit) [reply reqirements.dev:1]: its dev dependencies

Comment (wilmhit) [reply reqirements.dev]: I dont really like poetry, but sure we can use pipenv. Maybe next merge

**Review by songofroland — APPROVED** [review]

**Pull request merged**